src/Gaffer/MetadataAlgo.cpp

```diff
@@ -250,9 +264,15 @@ void setNumericBookmark( ScriptNode *scriptNode, int bookmark, Node *node )
 
 Node *getNumericBookmark( ScriptNode *scriptNode, int bookmark )
 {
-	// Return the first one we find. There should only ever be just one matching node.
+	// Return the first valid one we find. There should only ever be just one valid matching node.
 	for( Node *nodeWithMetadata : Metadata::nodesWithMetadata( scriptNode, numericBookmarkMetadataName( bookmark ), /* instanceOnly = */ true ) )
 	{
+		// Numeric bookmarks brought in by References will be ignored
```

Comment:
If we don't load numeric bookmarks into references (see suggestion above), then we wouldn't need to ignore them here, and we'd be properly enforcing our "there can be only one" rule, rather than just masking potential duplicates.

Reply:
Ok, I guess I was trying to preserve the referenced data as much as possible and opted for ignoring instead of removing. Not loading the metadata is going to clean up the code a fair bit, though. Will push an update later today that'll address all of your feedback. Thanks! :)